### DISCUSSION TITLE
Help with Map extractor

### DISCUSSION THREAD

**Discussion by helpfinder** [PiotrMachowski/Home-Assistant-custom-components-Xiaomi-Cloud-Map-Extractor#567] [Q&A]
Hi guys,
I have Xiaomi S20+ (EU) and I was able to add vacuum into Home Assistant via Xiaomi Miot Auto (Ha default Xiaomi integration was complaining about not supported device)
Then I moved firther with custom Xiaomi-Cloud-Map-Extractor and Xiaomi Vacuum Map and added Xiaomi Vacuum Map Camera into configuration.yaml

Problem I have is that in Dashboard in Vacuum map card I see error "Xiaomi Vacuum Map Camera" and in Picture entity Card nothing

![image](https://github.com/user-attachments/assets/fdd5ccf0-97a6-41fd-b128-d12a2e8b241d)

![image](https://github.com/user-attachments/assets/074e37b8-f25c-401a-b52d-8eb978028b49)

Home assistant System>Logs says

`Logger: homeassistant.helpers.entity
Source: helpers/entity.py:944
First occurred: 10:03:46 AM (154 occurrences)
Last logged: 10:16:29 AM

Update for camera.xiaomi_cloud_map_extractor fails
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 944, in async_update_ha_state
    await self.async_device_update()
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 1304, in async_device_update
    await hass.async_add_executor_job(self.update)
  File "/usr/local/lib/python3.12/concurrent/futures/thread.py", line 58, in run
    result = self.fn(*self.args, **self.kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/xiaomi_cloud_map_extractor/camera.py", line 278, in update
    self._handle_map_data(map_name)
  File "/config/custom_components/xiaomi_cloud_map_extractor/camera.py", line 335, in _handle_map_data
    map_data, map_stored = self._device.get_map(map_name, self._colors, self._drawables, self._texts,
                           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/xiaomi_cloud_map_extractor/common/vacuum.py", line 27, in get_map
    response = self.get_raw_map_data(map_name)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/xiaomi_cloud_map_extractor/common/vacuum.py", line 45, in get_raw_map_data
    map_url = self.get_map_url(map_name)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/xiaomi_cloud_map_extractor/common/vacuum_v2.py", line 18, in get_map_url
    if api_response is None or "result" not in api_response or "url" not in api_response["result"]:
                                                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
TypeError: argument of type 'NoneType' is not iterable`

Any help is appreciated

**Comment by horinka-gabor** [comment]
Hi. I got the same error with Roidmi Eve MAX, on latest HA.

**Comment by tiagofontoura** [comment]
I have an s20, I have the same problem

**Comment by helpfinder** [comment]
@PiotrMachowski any chance to review this?
Many thanks

**Comment by freeagent131** [comment]
Hi, is there any chance to get the map of this device?

xiaomi.vacuum.b106bk
<img src="https://github.com/user-attachments/assets/35286860-58ac-4eeb-8050-8c1621306822" width="500">

**Comment by alexsotoaguilera** [comment]
same on x10+
